src/vocab.rs

```diff
@@ -35,22 +35,29 @@ impl PyVocab {
         })
     }
 
-    fn ngram_indices(&self, word: &str) -> PyResult<Option<Vec<(String, usize)>>> {
+    fn ngram_indices(&self, word: &str) -> PyResult<PyObject> {
```

Comment:
Clippy complained about too complex return type with `PyResult<Option<Vec<(String, Option<usize>)>>>`. Returning `PyObject` should be simpler.

Reply:
I don't like it, since it makes it harder to just read the signatures and understand what a function does. IMO, Clippy is too overzealous in this case, since the type is still readable. Better to add an allow to silence Clippy on this one.